practicals/GNN_for_recommendations.ipynb

```diff
@@ -0,0 +1,1284 @@
+{
```

Comment:
<p>By this description, it's actually GAT, and not GCN?</p><p>From my reading of the code, it is in fact a GAT, where the attention weights are a function of only the edge weights, and not the incoming node features.</p><p> </p><p>This is important, since the attendees of the prac will only be familiar with GCN from the GNN tutorial, unless they did the extension sections. GATs form part of the extensions to the prac.</p><p> </p><p>Ideally, you should not use attention weights, and instead just use the edge weights directly. If you feel it's strictly necessary to use attention, then please add an extra explanation of the mechanism and refer the reader to the extended version of the GNN practical for more information.</p><p> </p><p>"with simple dense layer" -&gt; "with a simple dense layer"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2024/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Reply:
<p>It is indeed a GAT, i was thinking of just adding more explanation that because of attention weights this becomes GAT, but then thought this would make more complicated.. if you have GAT in your other prac, may be we could refer to that ?  or you rather suggest we just use the edge weights directly?</p>

Reply:
<p>I added an extra explanation.. please have a look :) </p>